src/robot/running/librarykeywordrunner.py

```diff
@@ -104,7 +104,14 @@ def _run_with_output_captured_and_signal_monitor(self, runner, context):
     def _run_with_signal_monitoring(self, runner, context):
         try:
             STOP_SIGNAL_MONITOR.start_running_keyword(context.in_teardown)
-            return runner()
+            runner_result = runner()
+            if (context.asynchronous.is_coroutine(runner_result) and not
+                context.asynchronous.is_loop_running()):
+                runner_result = (context
+                                 .asynchronous
+                                 .event_loop
+                                 .run_until_complete(runner_result))
```

Comment:
This looks a bit odd. Perhaps
```python
loop = context.asynchronous.event_loop
return loop.run_until_complete(runner_result)
```
would be better.

`context.asynchronous.event_loop` is also pretty long compared to earlier ´context.event_loop`, but having all the async stuff under `context.asynchronous` definitely has benefits as well. Probably best leave it like it is.